### PR TITLE
ply for python3 (distribution packages)

### PR DESCRIPTION
**Pull request by christianrauch**
Please add the following dependency to the rosdep database.

## Package name:

python3-ply

This is a follow up of https://github.com/ros/rosdistro/pull/31393 which added the pip version of this. This very PR adds the distributed python3 packages. I am renaming the old key to follow the pip convention, and reuse the "version 3" key for the distribution packages.

Distro packaging links:

## Links to Distribution Packages

- Debian: https://packages.debian.org/
  - https://packages.debian.org/bullseye/python3-ply
- Ubuntu: https://packages.ubuntu.com/
   - https://packages.ubuntu.com/focal/python3-ply
- Fedora: https://src.fedoraproject.org/browse/projects/
  - https://src.fedoraproject.org/rpms/python3-ply
- Arch: https://www.archlinux.org/packages/
  - https://archlinux.org/packages/extra/any/python-ply/
- Gentoo: https://packages.gentoo.org/
  - https://packages.gentoo.org/packages/dev-python/ply
- openSUSE:
  - https://software.opensuse.org/package/python3-ply

### CODE REVIEW

**Review by hidmic — COMMENTED** [review]
Overall LGTM. Would you mind adding the reference links for OpenSUSE and RHEL? Thanks!

**Comment by christianrauch** [comment]
> Would you mind adding the reference links for OpenSUSE and RHEL?

I added the openSUSE package path.

For searching RHEL, you will need an account or something. The CentOS 8 package is at: https://centos.pkgs.org/8/centos-baseos-x86_64/python3-ply-3.9-9.el8.noarch.rpm.html. The packages in CentOS and Fedora will have the same name.

File: rosdep/python.yaml [hunk 1]
```diff
@@ -7230,6 +7230,15 @@ python3-pkg-resources:
...
   rhel: ['python%{python3_pkgversion}-setuptools']
   ubuntu: [python3-pkg-resources]
+python3-ply:
+  arch: [python-ply]
+  debian: [python3-ply]
+  fedora: [python3-ply]
+  gentoo: [dev-python/ply]
+  opensuse: [python3-ply]
+  rhel:
+    "8": [python3-ply]
```
Comment (cottsay) [inline rosdep/python.yaml:7240]: Please use forward-looking syntax, as is the pattern used for the many other instances in rosdep where a package is available for the newest release but not older releases.

Also, please single-quote strings, as is the pattern used throughout rosdep.

```suggestion
    '*': [python3-ply]
    '7': null
```

**Review by cottsay — COMMENTED** [review]
> I am renaming the old key to follow the pip convention...

Typically renaming keys isn't something we do because it might break users in the wild. That said, the rule for `python-ply` is using the still unreleased OS_NAME wildcard feature in rosdep, so it is unlikely that there are widespread consumers.

> For searching RHEL, you will need an account or something.

In general, packages for any freely available RHEL clone will be identical, so CentOS, AlmaLinux, and Rocky Linux are all options for searching for RHEL package names.

**Comment by christianrauch** [comment]
@cottsay Is this good to go now?

**Review by cottsay — APPROVED** [review]
Fine with me, just need one more review from @ros/rosdeputies.

**Comment by christianrauch** [comment]
Ping... Can this go ahead?

**Review by clalancette — APPROVED** [review]

**Pull request merged**